libsolidity/formal/PredicateSort.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolidity/formal/PredicateSort.h>
+
+#include <libsolidity/formal/SMTEncoder.h>
+#include <libsolidity/formal/SymbolicTypes.h>
+
+using namespace std;
+using namespace solidity::util;
+using namespace solidity::smtutil;
+
+namespace solidity::frontend::smt
+{
+
+SortPointer interfaceSort(ContractDefinition const& _contract)
+{
+	return make_shared<smtutil::FunctionSort>(
+		stateSorts(_contract),
+		smtutil::SortProvider::boolSort
```

Comment:
I think `smtutil` is imported here
```suggestion
		SortProvider::boolSort
```

Reply:
I see here all over the place same thing. Would you prefer to leave as is?

Reply:
Here I would remove `smtutil` from Sorts, but need to keep it for `Expression` because that often gets confused with `frontend::Expression` by the compiler.